motionplan/ik/inverseKinematics.go

```diff
@@ -16,18 +19,79 @@ const (
 	defaultGoalThreshold = defaultEpsilon * defaultEpsilon
 )
 
-// InverseKinematics defines an interface which, provided with seed inputs and a Metric to minimize to zero, will output all found
+// InverseKinematics defines an interface which, provided with seed inputs and a function to minimize to zero, will output all found
 // solutions to the provided channel until cancelled or otherwise completes.
 type InverseKinematics interface {
-	referenceframe.Limited
-	// Solve receives a context, the goal arm position, and current joint angles.
-	Solve(context.Context, chan<- *Solution, []referenceframe.Input, StateMetric, int) error
+	// Solve receives a context, a channel to which solutions will be provided, a function whose output should be minimized, and a
+	// number of iterations to run.
+	Solve(context.Context, chan<- *Solution, []float64, func([]float64) float64, int) error
 }
 
 // Solution is the struct returned from an IK solver. It contains the solution configuration, the score of the solution, and a flag
 // indicating whether that configuration and score met the solution criteria requested by the caller.
 type Solution struct {
-	Configuration []referenceframe.Input
+	Configuration []float64
 	Score         float64
 	Exact         bool
 }
+
+// generateRandomPositions generates a random set of positions within the limits of this solver.
+func generateRandomPositions(randSeed *rand.Rand, lowerBound, upperBound []float64) []float64 {
+	pos := make([]float64, len(lowerBound))
+	for i, l := range lowerBound {
+		u := upperBound[i]
+
+		// Default to [-999,999] as range if limits are infinite
```

Comment:
why this range and not [-9999,9999]? 
why these numbers specifically?
[opt] 999 could probably be turned into a constant?

Reply:
This is essentially a random number. There's no reason for 999 or 9999 except that 999 is what we've always used. This function has simply moved from being a method on nlopt.

We need something to set as the limits in the event of infinity, and the actual value chosen should not matter very much as gradient descent should take care of it.

Agree this should be a constant.